packages/cli/lib/services/migration.service.ts

```diff
@@ -0,0 +1,132 @@
+import fs from 'fs';
+import path from 'path';
+import chalk from 'chalk';
+import { exec } from 'child_process';
+
+import { nangoConfigFile, loadLocalNangoConfig, determineVersion } from '@nangohq/shared';
+import { printDebug, getNangoRootPath } from '../utils.js';
+
+export const v1toV2Migration = async (loadLocation: string): Promise<void> => {
+    if (process.env['NANGO_CLI_UPGRADE_MODE'] === 'ignore') {
+        return;
+    }
+    const localConfig = await loadLocalNangoConfig(loadLocation);
+
+    if (!localConfig) {
+        return;
+    }
+
+    const version = determineVersion(localConfig);
+    if (version === 'v2') {
+        console.log(chalk.blue(`nango.yaml is already at v2.`));
+    }
+    if (version === 'v1' && localConfig.integrations) {
+        exec(`node ${getNangoRootPath()}/scripts/v1-v2.js ./${nangoConfigFile}`, (error) => {
+            if (error) {
+                console.log(chalk.red(`There was an issue migrating your nango.yaml to v2.`));
+                console.error(error);
+                return;
+            }
+            console.log(chalk.blue(`Migrated to v2 of nango.yaml!`));
+        });
+    }
+};
+
+async function createDirectory(dirPath: string, debug = false): Promise<void> {
+    if (fs.existsSync(dirPath)) {
+        if (debug) {
+            printDebug(`Directory already exists at ${dirPath}.`);
+        }
+        return;
+    }
+    try {
+        await fs.promises.mkdir(dirPath, { recursive: true });
+        if (debug) {
+            printDebug(`Created directory at ${dirPath}.`);
+        }
+    } catch (error) {
+        console.error(chalk.red(`There was an issue creating the directory at ${dirPath}.`), error);
```

Comment:
this seems to be a critical operation and if it fails everything else will fail. imo it should not be try/catch or stop the processing